stdlib/args.ysh

```diff
@@ -34,82 +34,76 @@
 #     - that's what it's doing, it's mutating "self"
 #     - _this is the thing we're currently creating
 
-proc Args ( ; out; ; block) {
+proc arg-parse (; place ; ; block_def) {
   ## Create an args spec which can be passed to parseArgs.
```

Comment:
Now that I see we have these `##` comments, I wonder if we do something like Elixir

https://hexdocs.pm/elixir/1.12/writing-documentation.html

And have a no-op `doc` verb, so it would be 

```
proc arg-parse {
  doc '''
  Multi-line 
  doc
  string.  And leading spaces are stripped according to the rule of multi-line strings
  '''
}
```

Right now we have

```
proc p {
  ### single line doc string
}
```

```
proc p {
  ### single line doc string
  ### this seems
  ### ugly
  ### and noisy

}
```